src/Netpbm.jl

```diff
@@ -156,62 +156,62 @@ end
     dat
 end
 
-function save(filename::File{format"PBMBinary"}, img; mapf=identity, mapi=nothing)
+function save(filename::File{format"PBMBinary"}, img; mapf=identity, mapi=nothing, comment=nothing)
     mapf = kwrename(:mapf, mapf, :mapi, mapi, :save)
     open(filename, "w") do s
         io = stream(s)
         write(io, "P4\n")
-        write(io, "# pbm file written by Julia\n")
+        write(io, comment === nothing ? "# pbm file written by Julia" : comment, "\n")
         save(s, img, mapf=mapf)
     end
 end
 
-function save(filename::File{format"PGMBinary"}, img; mapf=identity, mapi=nothing)
+function save(filename::File{format"PGMBinary"}, img; mapf=identity, mapi=nothing, comment=nothing)
     mapf = kwrename(:mapf, mapf, :mapi, mapi, :save)
     open(filename, "w") do s
         io = stream(s)
         write(io, "P5\n")
-        write(io, "# pgm file written by Julia\n")
+        write(io, comment === nothing ? "# pgm file written by Julia" : comment, "\n")
         save(s, img, mapf=mapf)
     end
 end
 
-function save(filename::File{format"PPMBinary"}, img; mapf=identity, mapi=nothing)
+function save(filename::File{format"PPMBinary"}, img; mapf=identity, mapi=nothing, comment=nothing)
     mapf = kwrename(:mapf, mapf, :mapi, mapi, :save)
     open(filename, "w") do s
         io = stream(s)
         write(io, "P6\n")
-        write(io, "# ppm file written by Julia\n")
+        write(io, comment === nothing ? "# ppm file written by Julia" : comment, "\n")
         save(s, img, mapf=mapf)
     end
 end
 
-function save(filename::File{format"PBMText"}, img; mapf=identity, mapi=nothing)
+function save(filename::File{format"PBMText"}, img; mapf=identity, mapi=nothing, comment=nothing)
     mapf = kwrename(:mapf, mapf, :mapi, mapi, :save)
     open(filename, "w") do s
         io = stream(s)
         write(io, "P1\n")
-        write(io, "# pbm file written by Julia\n")
+        write(io, comment === nothing ? "# pbm file written by Julia" : comment, "\n")
         save(s, img, mapf=mapf)
     end
 end
 
-function save(filename::File{format"PGMText"}, img; mapf=identity, mapi=nothing)
+function save(filename::File{format"PGMText"}, img; mapf=identity, mapi=nothing, comment=nothing)
     mapf = kwrename(:mapf, mapf, :mapi, mapi, :save)
```

Comment:
A dispatch design comment, maybe:

```julia
_read_metadata_comment(x::AbstractArray) = nothing
_read_metadata_comment(x::ImageMeta) = ...

function save(filename::File{format"PGMText"}, img; mapf=identity, mapi=nothing, comment=nothing)
    comment = _read_metadata(img)
    ...
    write(io, isnothing(comment) ? "# pbm file written by Julia" : comment, "\n")
    ...
end
```

By orthogonally designing the dispatch, we no longer need the 6 additional new save/load methods.

A style comment: I personally prefer `isnothing(x)` to `x === nothing`.

Reply:
Thanks for the comment on dispatch, I'll update to your cleaner suggestion.
I got used to using `===nothing` after reading comments that it was faster, but I have no strong feelings on the matter, and will change that, too.